bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/navigation/NavigationImpl.java

```diff
@@ -174,12 +174,8 @@ private void expandCatalogRoot(Page catalogPage, List<NavigationItem> pages) {
 
         Integer rootCategoryId = readPageConfiguration(catalogPage, PN_MAGENTO_ROOT_CATEGORY_ID);
         if (rootCategoryId == null) {
-            ConfigurationBuilder configBuilder = catalogPage.adaptTo(ConfigurationBuilder.class);
-
-            if (configBuilder != null) {
-                ValueMap properties = configBuilder.name("cloudconfigs/commerce").asValueMap();
-                rootCategoryId = properties.get(PN_MAGENTO_ROOT_CATEGORY_ID, Integer.class);
-            }
+            ComponentsConfiguration properties = catalogPage.adaptTo(ComponentsConfiguration.class);
```

Comment:
Will that work? (catalogPage is not a Resource)

Reply:
It probably won't. I didn't notice it in the tests because it always falls back to the context configuration.

Reply:
Actually that component reads the configuration in the "opposite" order of what we want to do: it should first get the config from that new `ComponentsConfiguration` and then fallback to the old way ... that's probably why the test didn't fail.

Reply:
Yes, this is the way it's intended for this component - it first reads the category from the page properties and then it falls back to the configuration.